test/sshkit/scp_functional_test.exs

```diff
@@ -1,3 +1,178 @@
 defmodule SSHKit.SCPFunctionalTest do
   use SSHKit.FunctionalCase, async: true
+
+  alias SSHKit.SCP
+  alias SSHKit.SSH
+
+  @defaults [silently_accept_hosts: true]
+
+  describe "upload/4" do
+    @tag boot: 1
+    test "sends a file", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace/local_file.txt"
+      remote = "file.txt"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote)
+      assert verify_transfer(conn, local, remote)
+      :ok = SSH.close(conn)
+    end
+
+    @tag boot: 1
+    test "recursive: true", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace"
+      remote = "/home/#{host.user}/destination"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote, recursive: true)
+      assert verify_transfer(conn, local, remote)
+      :ok = SSH.close(conn)
+    end
+
+    @tag boot: 1
+    test "preserve: true", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace/local_file.txt"
+      remote = "file.txt"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote, preserve: true)
+      assert verify_mode(conn, local, remote)
+      # assert verify_atime(conn, local, remote)
```

Comment:
> The `atime` assertions are commented out because there seems to be a situation where although we are setting `atime` to the correct value (I have debugged this a little to verify) the compared values are always off by just a few seconds between the remote file(s) and the local file(s). I'm not sure if there is a reliable way to test this assertion, but maybe you have some ideas.

i've just seen your comment above. is it still happening or did you forget to remove the comment? I will try to locally reproduce the error.